openedx/openedx-platform
# Display tag counts on course outline [FC-0036]

## Description

When the "tagging/taxonomy" feature flag is on, this updates the course outline page to show the count of tags that are applied to each unit. The data for all units is loaded at once, rather than each unit at a time.

## Supporting information

This implements https://github.com/openedx/modular-learning/issues/118 part 1b: "For units which have tags, the number of tags that they have appears next to this tag icon. Clicking the number/icon will also open the tag drawer."

## Testing instructions

1. Enable the `new_studio_mfe.use_tagging_taxonomy_list_page` waffle flag
2. Load the taxonomy sample data, ideally the version from [this PR](https://github.com/open-craft/taxonomy-sample-data/pull/3) which is more random.
3. View the course outline page for one of the taxonomy sample courses. It should look like the following. Note the tag counts shown.

![Screenshot 2023-11-10 at 11 08 42 AM](https://github.com/openedx/edx-platform/assets/945577/51894fca-0f32-4a60-8b6d-8b00be44eb12)


## Deadline

None

## Other information

Depends on:
- https://github.com/openedx/openedx-learning/pull/115
- https://github.com/openedx/frontend-app-course-authoring/pull/654 (not required for this to merge though, because of waffle flag)

Private ref: [FAL-3564](https://tasks.opencraft.com/browse/FAL-3564).
